File: Clojure/problem036.clj
#! /usr/bin/env clojure
; Problem 36 - Project Euler
; http://projecteuler.net/index.php?section=problems&id=36
(require '[clojure.string :as s])

(defn palindromic10? [x]
  (let [strx (str x)]
    (= strx (s/reverse strx))))

(defn palindromicbin? [x]
  (let [binx (Integer/toBinaryString x)]
       (= binx (s/reverse binx))))

(defn palindromic2bases? [x]
  (and (palindromic10? x) (palindromicbin? x)))

(->> (range 1 1000000)
     (filter palindromic2bases?)
     (reduce +)
     (println))

File: Kotlin/problem036.kt
// Problem 36 - Project Euler
// http://projecteuler.net/index.php?section=problems&id=36
fun Int.is_pal(base: Int): Boolean {
    return this.toString(base).reversed() == this.toString(base)
}

fun main() {
    println((1..1_000_000 step 2)
            .filter { it.is_pal(2) && it.is_pal(10) }
            .sum())
}
